chalice/config.py

```diff
@@ -151,16 +151,13 @@ def lambda_python_version(self) -> str:
         if major == 2:
             return 'python2.7'
         # Python 3 for backwards compatibility needs to select python3.6
-        # for python versions 3.0-3.6. 3.7 and higher will use python3.7.
+        # for python versions 3.0-3.6. 3.7-3.10 will use their version.
+        # 3.11 and higher will use 3.11
         elif (major, minor) <= (3, 6):
             return 'python3.6'
```

Comment:
@jamesls I was noticing python 3.6 appears no longer supported: https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html

Wasn't sure about changing this to force 3.7 as minimum?  Maybe removing 3.6 would be better as a separate PR if it's wanted?...